File: tests/xcalablemp/LARGE-SIZE-TESTS/global-view/gmove/F/gmove_G2L_1a6t_bc.f90
program tgmove
integer,parameter :: n=4
integer a(n,n,n,n,n,n), b(n,n,n,n,n,n)
integer xmp_node_num
!$xmp nodes p(2,2,2,2,2,2)
!$xmp template tx(n,n,n,n,n,n)
!$xmp distribute tx(cyclic(2),cyclic(2),cyclic(2),cyclic(2),cyclic(2),cyclic(2)) onto p
!$xmp align a(*,*,*,*,*,i5) with tx(*,*,*,*,*,i5)

!$xmp loop (i5) on tx(*,*,*,*,*,i5)
do i5=1,n
  do i4=1,n
    do i3=1,n
      do i2=1,n
        do i1=1,n
          do i0=1,n
            a(i0,i1,i2,i3,i4,i5)=i0+i1+i2+i3+i4+i5
          end do
        end do
      end do
    end do
  end do
end do

do i5=1,n
  do i4=1,n
    do i3=1,n
      do i2=1,n
        do i1=1,n
          do i0=1,n
            b(i0,i1,i2,i3,i4,i5)=0
          end do
        end do
      end do
    end do
  end do
end do

!$xmp gmove
b(2:n,2:n,2:n,2:n,2:n,2:n)=a(2:n,2:n,2:n,2:n,2:n,2:n)

ierr=0
do i5=2,n
  do i4=2,n
    do i3=2,n
      do i2=2,n
        do i1=2,n
          do i0=2,n
            ierr=ierr+abs(b(i0,i1,i2,i3,i4,i5)-i0-i1-i2-i3-i4-i5)
          end do
        end do
      end do
    end do
  end do
end do

!$xmp reduction (max:ierr)
call chk_int(ierr)

stop
end program tgmove

File: tests/xcalablemp/LARGE-SIZE-TESTS/global-view/gmove/C/gmove_G2L_1a6t_bc.c
#define N 2
#include <stdio.h>
#include <math.h>

extern int chk_int(int ierr);

int n=N;
double a[n][n][n][n][n][n];
#pragma xmp nodes p(2,2,2,2,2,2)
#pragma xmp template tx(0:n-1,0:n-1,0:n-1,0:n-1,0:n-1,0:n-1)
#pragma xmp distribute tx(cyclic(2),cyclic(2),cyclic(2),cyclic(2),cyclic(2),cyclic(2)) onto p
#pragma xmp align a[*][*][*][*][*][i] with tx(*,*,*,*,*,i)

int main(){

  int i0,i1,i2,i3,i4,i5,ierr;
  double b[N][N][N][N][N][N],err;

  for(i5=0;i5<n;i5++){
    for(i4=0;i4<n;i4++){
      for(i3=0;i3<n;i3++){
        for(i2=0;i2<n;i2++){
          for(i1=0;i1<n;i1++){
#pragma xmp loop (i0) on tx(*,*,*,*,*,i0)
            for(i0=0;i0<n;i0++){
              a[i5][i4][i3][i2][i1][i0]=i5+i4+i3+i2+i0+i1+1;
            }
          }
        }
      }
    }
  }

  for(i5=0;i5<n;i5++){
    for(i4=0;i4<n;i4++){
      for(i3=0;i3<n;i3++){
        for(i2=0;i2<n;i2++){
          for(i1=0;i1<n;i1++){
            for(i0=0;i0<n;i0++){
              b[i5][i4][i3][i2][i1][i0]=0.0;
            }
          }
        }
      }
    }
  }

#pragma xmp gmove
  b[1:n-1][1:n-1][1:n-1][1:n-1][1:n-1][1:n-1]=a[1:n-1][1:n-1][1:n-1][1:n-1][1:n-1][1:n-1];

  err=0.0;
  for(i5=1;i5<n;i5++){
    for(i4=1;i4<n;i4++){
      for(i3=1;i3<n;i3++){
        for(i2=1;i2<n;i2++){
          for(i1=1;i1<n;i1++){
            for(i0=1;i0<n;i0++){
              err=err+fabs(b[i5][i4][i3][i2][i1][i0]-i5-i4-i3-i2-i0-i1-1);
            }
          }
        }
      }
    }
  }

#pragma xmp reduction (MAX:err)
  ierr=err;
  chk_int(ierr);
}
